_posts/2020-12-01-infrastructure-migration-finished.md

```diff
@@ -0,0 +1,275 @@
+This is not okay, in no means what so ever.
+That is a disaster waiting to happen .. it is no longer an "if", but a "when".
+
+So, you say, the problem is that you are maintaining this by yourself?
+Find someone else to help out, and the problem is solved too!
+
+Yes, yes, yes, thousand times yes.
+That would really help.
+But .. I have been doing this for 16 years .. I have had many people come by and say: I am going to help you.
+When they dive in a bit deeper, they either realise the immense amount of work that would need to be done, or that it is very repetitive work (login to the system every month, update all the software, validate everything is still operating correctly, log off).
+To give a few examples:
+- BaNaNaS was going to be replaced with [BaNaNaS v2](https://github.com/`frosch`123/bananas2).
+  I believe the first talk about this was somewhere in 2014.
+  It would solve many issues with BaNaNaS v1.
+  It was very ambitious.
+  In 2017 it was given another go.
+  It stalled a few months after.
+  There is some initial code (code we reused for BaNaNaS v1.5, more on that later), but no much more than that.
+- [MediaWiki](https://www.mediawiki.org/wiki/MediaWiki) was running a very old version, and several people said they were going to help upgrading it to a newer.
+  We supplied VMs with the current version, so people could experiment with it.
+  I never heard back from ANY of them.
+  Like .. nothing after the initial offer and talk.
+
+There are many more examples, but I am sure you are understanding by now where I am coming from.
+And do not get me wrong: all of those people who offered to help, they did it with the best intentions, and I thank them for reaching out.
+But creating software and maintaining infrastructure is hard and difficult.
+It is not something you do on a Sunday afternoon and be done with it.
+It takes serious time and commitment.
+And myself running towards my 40s, time is not something that is ample anymore.
+It was when were were in our 20s, and the university allowed many many many many hours of doing this crazy things.
+
+So, in 2018, we decided this had to change.
+A few drastic changes were made, and many more would follow in the next two years.
+
+The big thing to solve: so far we have been using what is called unmanaged hosting.
+This means we get access to a server, and after that we have to do everything ourself.
+This is very cheap, and the main reason we used to do this; but it also means you have to put in the hours to maintain it.
+As pointed out above .. this was not happening.
+So we wanted to switch to managed hosting.
+This is a lot more expensive, but also would mean we didn't have to put in the work.
+With managed hosting, the provider is taking care of many things:
+
+- OS updates
+- Security patching
+- If you use a [SaaS (Software-as-a-Service)](https://en.wikipedia.org/wiki/Software_as_a_service), keeping the application up-to-date
```

Comment:
Swap sentence parts, almost reads as an incomplete sentence otherwise
```suggestion
- Keeping the application up-to-date, if you're using a [SaaS (Software-as-a-Service)](https://en.wikipedia.org/wiki/Software_as_a_service)
```